web/client/actions/layerFilter.js

```diff
@@ -62,3 +65,19 @@ export function initLayerFilter(filter) {
     };
 }
 
+export function layerFilterByLegend(layerId, nodeType, legendCQLFilter) {
+    return {
+        type: LAYER_FILTER_BY_LEGEND,
+        legendCQLFilter,
+        nodeType,
+        layerId
+    };
+}
+
+export function resetLegendFilter(reason, value) {
+    return {
+        type: RESET_LAYER_FILTER_BY_LEGEND,
+        reason,     // here the reason for reset is change 'style' or change the enable/disable interactive legend config 'disableEnableInteractiveLegend'
+        value
+    };
+}
```

Comment:
Probably also this function could be removed and create a utils that update the layerFilter removing the filter part related to legend. We need to reset the legend in following places:

- when we disable the interactive from settings. There the onChange should be available
- when we change the style. Also in this case we should use the the updateNode action passed as onUpdateNode